library/init/lean/elaborator.lean

```diff
@@ -930,6 +930,41 @@ let matchesOnly := match spec.only with
 | some only := n = only.id.val ∨ only.ids.any (λ id, n = id.val) in
 if matchesOnly then some (spec.id.val ++ n) else none
 
+/-- Global name lookup
+
+Accessible global name:
+A name `n` is called an *accessible global name* if the environment contains a
+declaration `n` or `_private.<module>.n` where `<module>` is the name of the
+current module.
```

Comment:
Note that I've decided to document a simpler `private` mechanism here: a private name is visible within its entire module, so we do not need any new state. What do you think?

Reply:
I think it is fine. It matches the behavior of `static`/`private` declarations in other languages.

Reply:
Note that currently private names also include an index, so this approach would not work. However, I believe the index is only there so that currently one *can* declare the same private name in different sections of the same file. Is that correct?

Reply:
Yes, this is correct.